internal/kuberesource/parts.go

```diff
@@ -217,7 +228,12 @@ type CoordinatorConfig struct {
 }
 
 // Coordinator constructs a new CoordinatorConfig.
-func Coordinator(namespace string) *CoordinatorConfig {
+func Coordinator(namespace string, platform platforms.Platform) (*CoordinatorConfig, error) {
+	runtimeHandler, err := manifest.DefaultPlatformHandler(platform)
+	if err != nil {
+		return nil, fmt.Errorf("getting default runtime handler: %w", err)
+	}
```

Comment:
The coordinator resource does not really depend on the platform, so I'd prefer not to have the `platform` argument and to patch the `runtimeClassName` outside of this constructor. Then you would not have to pass down the argument through all functions.

Same goes for all the other resources (openssl etc.).

Reply:
What'd be the benefit of this though? In reality, the resource won't work without the runtime handler, so it does depend on it in some sense?

Reply:
The resources work without a runtime handler, with the kata-cc runtime handler, with older runtime handlers, etc. Imho, the runtime handler is a property  of the environment and should thus be added where the environment is set up, i.e. in test runners etc. And this would have the upside that you don't need to add and maintain these platform arguments everywhere.

Reply:
Sounds reasonable. Where would you implement the patching?

Reply:
In the resourcegen binary and the test runners, probably?